metaflow/plugins/airflow/airflow.py

```diff
@@ -0,0 +1,670 @@
+import base64
+import json
+import os
+import random
+import string
+import sys
+from datetime import datetime, timedelta
+
+import metaflow.util as util
+from metaflow.decorators import flow_decorators
+from metaflow.exception import MetaflowException
+from metaflow.metaflow_config import (
+    BATCH_METADATA_SERVICE_HEADERS,
+    BATCH_METADATA_SERVICE_URL,
+    DATASTORE_CARD_S3ROOT,
+    DATASTORE_SYSROOT_S3,
+    DATATOOLS_S3ROOT,
+    KUBERNETES_SERVICE_ACCOUNT,
+)
+from metaflow.parameters import deploy_time_eval
+from metaflow.plugins.kubernetes.kubernetes import Kubernetes
+
+# TODO: Move chevron to _vendor
+from metaflow.plugins.cards.card_modules import chevron
+from metaflow.plugins.timeout_decorator import get_run_time_limit_for_task
+from metaflow.util import dict_to_cli_options, get_username, compress_list
+from metaflow.parameters import JSONTypeClass
+
+from . import airflow_utils
+from .exception import AirflowException
+from .sensors import SUPPORTED_SENSORS
+from .airflow_utils import (
+    AIRFLOW_TASK_ID_TEMPLATE_VALUE,
+    RUN_ID_LEN,
+    RUN_ID_PREFIX,
+    TASK_ID_XCOM_KEY,
+    AirflowTask,
+    Workflow,
+)
+from metaflow import current
+
+AIRFLOW_DEPLOY_TEMPLATE_FILE = os.path.join(os.path.dirname(__file__), "dag.py")
+
+
+class Airflow(object):
+
+    parameter_macro = "{{ params | json_dump }}"
+
+    task_id = AIRFLOW_TASK_ID_TEMPLATE_VALUE
```

Comment:
Why wouldn't you directly use `AIRFLOW_TASK_ID_TEMPLATE_VALUE`. Maybe rename it to `AIRFLOW_TASK_ID`

Reply:
Can be refactored.